src/Compiler/Checking/CheckExpressions.fs

```diff
@@ -6650,12 +6650,45 @@ and TcCtorCall isNaked cenv env tpenv (overallTy: OverallTy) objTy mObjTyOpt ite
     if isInterfaceTy g objTy then
         error(Error((if superInit then FSComp.SR.tcInheritCannotBeUsedOnInterfaceType() else FSComp.SR.tcNewCannotBeUsedOnInterfaceType()), mWholeCall))
 
+    let rec getConstructorArgs expr =
+        match expr with
+        | SynExpr.Paren(expr = SynExpr.App(funcExpr = expr)) -> getConstructorArgs expr
+        | SynExpr.Paren(expr = SynExpr.Tuple(exprs = synExprs)) -> synExprs |> List.collect getConstructorArgs
+        | SynExpr.App(funcExpr = expr1; argExpr = expr2) -> [ expr1 ; expr2 ] |> List.collect getConstructorArgs
+        | SynExpr.Ident(id) -> [ id ]
+        | _ -> []
+
     match item, args with
-    | Item.CtorGroup(methodName, minfos), _ ->
+    | Item.CtorGroup(methodName, minfos), argExprs ->
         let meths = List.map (fun minfo -> minfo, None) minfos
         if isNaked && TypeFeasiblySubsumesType 0 g cenv.amap mWholeCall g.system_IDisposable_ty NoCoerce objTy then
             warning(Error(FSComp.SR.tcIDisposableTypeShouldUseNew(), mWholeCall))
 
+        let ctorArgs = argExprs |> List.collect getConstructorArgs
```

Comment:
I wonder if this should be a separate method akin to `CheckRequiredProperties`.
We can reuse it in `TcMethodApplication`, when we handle post-hoc property assignments.

Reply:
Done. See my last commit